In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import skew
import scipy.stats as st

### **Ana: Educación(Q4)]**

In [95]:
df_formacion=pd.read_csv('df_a_usar_2.csv')

In [96]:
df_formacion['education_q4'].value_counts()

Master’s degree                                                      10132
Bachelor’s degree                                                     9907
Doctoral degree                                                       2795
Some college/university study without earning a bachelor’s degree     1735
I prefer not to answer                                                 627
No formal education past high school                                   417
Professional doctorate                                                 360
Name: education_q4, dtype: int64

In [97]:
df_formacion['education_q4'].count()
#no tenemos ni nulos ni registros vacios para formación

25973

In [6]:
df_formacion.duplicated().sum()

0

In [ ]:
df_formacion.isnull().sum()

### Nos quedamos con las columnas que seleccionamos en el grupo para las preguntas, que no tiene duplicados con un total de 25973 registros

In [ ]:
df_def_fomacion=df_formacion.loc[:,[ 'index', 'time', 'age', 'gender','country_q3','education_q4','role_q5','experience_q6','sector_q20','company_size_q21','anual_income_q25']]
df_def_fomacion

In [ ]:
plt.figure(figsize =(10,8))
sns.histplot(x=df_formacion['education_q4'],data = df_formacion, hue='gender', kde=True)
plt.xticks(rotation= 75)
plt.show()

In [10]:
df_def_fomacion.isnull().sum()

index                   0
time                    0
age                     0
gender                  0
country_q3              0
education_q4            0
role_q5                 0
experience_q6           0
sector_q20           9648
company_size_q21     9722
anual_income_q25    10582
dtype: int64

In [11]:
#ed_gen=pd.DataFrame(df_def_fomacion.groupby(['education_q4'])['gender'].value_counts())
#ed_gen

In [12]:
#ed_gen.rename(columns={'gender':'frecuencia'}, inplace= True)
#ed_gen.reset_index()

In [30]:
def porcentaje(df):
    array2=[]
    for i in df.columns:
        for v in df[i]:
            print(i)
        z=v*100/df[i].sum()
        array2.append(z)
        
    return  np.round(np.array(array2), decimals=2)
    
    #funcion para hallar porcentajes
    #print(z)
    #tamaño_genero["Woman%"] =tamaño_genero_porcentaje (tamaño_genero["'Woman"].apply(porcentaje))

In [ ]:
for col in df.columns:
    print("------------------------------------------------------------", "\n")
    print(f"En la columna {col} tenemos los siguientes valores unicos")
    z=col*100/df[col].sum()
    array2.append(z)
    print(df[col].unique())
    print("\n", "
      return  np.round(np.array(array2), decimals=2)

In [ ]:
for col in categoricas.columns:
    print("------------------------------------------------------------", "\n")
    print(f"En la columna {col} tenemos los siguientes valores unicos")
    print(pd.DataFrame(df[col].value_counts()))
    print("\n", "------------------------------------------------------------", "\n")

In [ ]:
educacion_gen2=pd.DataFrame(pd.crosstab(df_formacion['gender'],df_formacion['education_q4'], margins=True))
educacion_gen2
#'Man':'Bachelor’s degree'31.7%, 'Doctoral degree'14.02%,'I prefer not to answer':1.9%, 'Master’s degree'43.8%,
#'No formal education past high school'1.6%, 'Professional doctorate'1.76%,'Some college/university study without earning a bachelor’s 
# degree'5%
#universitarios con titulo:88.65 hombres, 90.6 mujeres, 80.31 resto
#'Woman''Bachelor’s degree'28.2%, 'Doctoral degree'16.07%,'I prefer not to answer':2.8%, 'Master’s degree'45.8%,
#'No formal education past high school'0.7%, 'Professional doctorate'2.45%,'Some college/university study without earning a bachelor’s 
# degree'3.8%
#'Resto':'Bachelor’s degree'27.8, 'Doctoral degree'12.16,'I prefer not to answer':8.6%, 'Master’s degree'38.5%,
#'No formal education past high school'2.68%, 'Professional doctorate'1.85%,'Some college/university study without earning a bachelor’s 
# degree'8%


In [ ]:
educacion_gen2_no_null=pd.DataFrame(pd.crosstab(df_formacion_sin_inactivos['gender'],df_formacion_sin_inactivos['education_q4'], margins=True))
educacion_gen2_no_null

In [ ]:
plt.figure(figsize =(10,8))
sns.histplot(x=df_def_fomacion['education_q4'],data = df_def_fomacion, hue='gender')#, kde=True)
plt.xticks(rotation= 75)
plt.show()

In [ ]:
plt.figure(figsize =(10,8))
sns.histplot(x=df_formacion_sin_inactivos['education_q4'],data = df_formacion_sin_inactivos, hue='gender')#, kde=True)
plt.xticks(rotation= 75)
plt.show()

In [ ]:
educacion_gen3=pd.DataFrame(pd.crosstab(df_def_fomacion['role_q5'],df_def_fomacion['education_q4'], margins=True).reset_index())
educacion_gen3

In [ ]:
educacion_gen3_no_null=pd.DataFrame(pd.crosstab(df_formacion_sin_inactivos['role_q5'],df_formacion_sin_inactivos['education_q4'], margins=True).reset_index())
educacion_gen3_no_null

In [ ]:
plt.figure(figsize =(10,8))
sns.histplot(x=df_def_fomacion['role_q5'],data = df_def_fomacion, hue='education_q4')#, kde=True)
plt.xticks(rotation= 75)
plt.show()

In [ ]:
plt.figure(figsize =(10,8))
sns.histplot(x=df_formacion_sin_inactivos['role_q5'],data = df_formacion_sin_inactivos, hue='education_q4')#, kde=True)
plt.xticks(rotation= 75)
plt.show()

In [ ]:
educacion_gen4=pd.DataFrame(pd.crosstab(df_def_fomacion['anual_income_q25'],df_def_fomacion['education_q4'], margins=True))
educacion_gen4

In [ ]:
educacion_gen4_no_null=pd.DataFrame(pd.crosstab(df_formacion_sin_inactivos['anual_income_q25'],df_formacion_sin_inactivos['education_q4'], margins=True))
educacion_gen4_no_null

In [ ]:
plt.figure(figsize =(10,8))
sns.histplot(x=df_def_fomacion['anual_income_q25'],data = df_def_fomacion, hue='education_q4')#, kde=True)
plt.xticks(rotation= 75)
plt.show()

In [ ]:
plt.figure(figsize =(10,8))
sns.histplot(x=df_formacion_sin_inactivos['anual_income_q25'],data = df_formacion_sin_inactivos, hue='education_q4')#, kde=True)
plt.xticks(rotation= 75)
plt.show()

In [ ]:
educacion_gen5=pd.DataFrame(pd.crosstab(df_def_fomacion['company_size_q21'],df_def_fomacion['education_q4'], margins=True).reset_index())
educacion_gen5

In [ ]:
educacion_gen5_no_null=pd.DataFrame(pd.crosstab(df_formacion_sin_inactivos['company_size_q21'],df_formacion_sin_inactivos['education_q4'], margins=True).reset_index())
educacion_gen5_no_null

In [ ]:
plt.figure(figsize =(10,8))
sns.histplot(x=df_def_fomacion['education_q4'],data = df_def_fomacion, hue='company_size_q21')#, kde=True)
plt.xticks(rotation= 75)
plt.show()

In [ ]:
plt.figure(figsize =(10,8))
sns.histplot(x=df_formacion_sin_inactivos['education_q4'],data = df_formacion_sin_inactivos, hue='company_size_q21')#, kde=True)
plt.xticks(rotation= 75)
plt.show()

In [ ]:
educacion_gen6=pd.DataFrame(pd.crosstab(df_def_fomacion['country_q3'],df_def_fomacion['education_q4'], margins=True).reset_index())
educacion_gen6

In [ ]:
educacion_gen6_no_null=pd.DataFrame(pd.crosstab(df_formacion_sin_inactivos['country_q3'],df_formacion_sin_inactivos['education_q4'], margins=True).reset_index())
educacion_gen6_no_null

In [ ]:
plt.figure(figsize =(20,15))
sns.histplot(x=df_def_fomacion['country_q3'],data = df_def_fomacion, hue='education_q4')#, kde=True)
plt.xticks(rotation= 75)
plt.show()

In [ ]:
plt.figure(figsize =(20,15))
sns.histplot(x=df_formacion_sin_inactivos['country_q3'],data = df_formacion_sin_inactivos, hue='education_q4')#, kde=True)
plt.xticks(rotation= 75)
plt.show()

In [ ]:
educacion_gen7=pd.DataFrame(pd.crosstab(df_def_fomacion['experience_q6'],df_def_fomacion['education_q4'], margins=True))
educacion_gen7

In [ ]:
educacion_gen7_no_null=pd.DataFrame(pd.crosstab(df_formacion_sin_inactivos['experience_q6'],df_formacion_sin_inactivos['education_q4'], margins=True))
educacion_gen7_no_null

Los datos ue menos varian segun empleabilidad y experiencia, son los de los doctorados

In [ ]:
plt.figure(figsize =(10,8))
sns.histplot(x=df_def_fomacion['education_q4'],data = df_def_fomacion, hue='experience_q6')#, kde=True)
plt.xticks(rotation= 75)
plt.show()

In [ ]:
plt.figure(figsize =(10,8))
sns.histplot(x=df_formacion_sin_inactivos['education_q4'],data = df_formacion_sin_inactivos, hue='experience_q6')#, kde=True)
plt.xticks(rotation= 75)
plt.show()

In [ ]:
educacion_gen8=pd.DataFrame(pd.crosstab(df_def_fomacion['sector_q20'],df_def_fomacion['education_q4'], margins=True).reset_index())
educacion_gen8

In [ ]:
educacion_gen8_no_null=pd.DataFrame(pd.crosstab(df_formacion_sin_inactivos['sector_q20'],df_formacion_sin_inactivos['education_q4'], margins=True).reset_index())
educacion_gen8_no_null

In [ ]:
plt.figure(figsize =(10,8))
sns.histplot(x=df_def_fomacion['sector_q20'],data = df_def_fomacion, hue='education_q4')#, kde=True)
plt.xticks(rotation= 75)
plt.show()

In [ ]:
plt.figure(figsize =(10,8))
sns.histplot(x=df_formacion_sin_inactivos['sector_q20'],data = df_formacion_sin_inactivos, hue='education_q4')#, kde=True)
plt.xticks(rotation= 75)
plt.show()

In [ ]:
educacion_gen9=pd.DataFrame(pd.crosstab(df_def_fomacion['age'],df_def_fomacion['education_q4'], margins=True).reset_index())
educacion_gen9

In [ ]:
educacion_gen9_no_null=pd.DataFrame(pd.crosstab(df_formacion_sin_inactivos['age'],df_formacion_sin_inactivos['education_q4'], margins=True).reset_index())
educacion_gen9_no_null

In [ ]:
plt.figure(figsize =(10,8))
sns.histplot(x=df_def_fomacion['education_q4'],data = df_def_fomacion, hue='age')#, kde=True)
plt.xticks(rotation= 75)
plt.show()

In [ ]:
plt.figure(figsize =(10,8))
sns.histplot(x=df_formacion_sin_inactivos['education_q4'],data = df_formacion_sin_inactivos, hue='age')
plt.xticks(rotation= 75)
plt.show()



En general, a mayor edad, mayor formacion, los doctorados, van bajando la proporcion con la edad, a la vez que va subiendo la licenciatura casi. El grado de master, mas o menos se mantiene, an los activos.